src/main/java/com/facebook/openwifirrm/modules/ApiServer.java

```diff
@@ -269,16 +290,23 @@ private String getFullUrl(String path, String queryString) {
 	 * HTTP 403 response and return false.
 	 */
 	private boolean performOpenWifiAuth(Request request, Response response) {
-		// TODO check if request came from internal endpoint
-		boolean internal = true;
-		String internalName = request.headers("X-INTERNAL-NAME");
-		if (internal && internalName != null) {
-			// Internal request, validate "X-API-KEY"
-			String apiKey = request.headers("X-API-KEY");
-			if (apiKey != null && apiKey.equals(serviceKey)) {
-				// auth success
-				return true;
+		boolean internal = request.port() == params.internalHttpPort;
```

Comment:
Just being paranoid here (even though this will probably never happen), also check if port is positive?
```java
int port = request.port();
boolean internal = port > 0 && port == params.internalHttpPort;
```